pkg/api/node.go

```diff
@@ -94,6 +95,13 @@ func (m *nodeMetrics) nodes(ctx context.Context, options *metainternalversion.Li
 	if options != nil && options.LabelSelector != nil {
 		labelSelector = options.LabelSelector
 	}
+	requirement, err := labels.NewRequirement("metrics-server-skip", selection.DoesNotExist, []string{})
```

Comment:
Not very convinced that this solves original issue. As adding a new label to nodes is not as trivial as you think. It's not like user will iterate through all their nodes once add the label. Most K8s clusters are not dynamic  with node auto provisioning. Updating labels on nodes each time an alert is triggered is not maintainable. 

Don't think that `AWS EKS Fargate` or any other provisioner of non-standard K8s nodes will adapt the label you proposed soon.

Alternative option would be to allow users to provide the label selector by themselves. Don't have experience with `AWS EKS Fargate` so it might be worth asking  on https://github.com/kubernetes-sigs/metrics-server/issues/1179, if there are already specific label for those nodes. 

I expect that we can add a flag `--node-selector` and accept syntax similar to kubectl https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering

From `kubectl get --help` 
```
    -l, --selector='':
	Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2). Matching
	objects must satisfy all of the specified label constraints.
``

For example `--node-selector='fargate!=true'`
cc @sergeyshevch for feedback

Reply:
Creating a flag will be perfect. 

My nodes were created with karpenter and it automatically has a label eks.amazonaws.com/compute-type=fargate
Assigning a fixed label to nodes is possible but it's a harder way. (At least I need to create separate karpenter provisioner for this

Maybe we also need to be sure that the user can specify multiple labels. And nodes, when at least one selector is satisfied, will be excluded

Reply:
Yeah, I think  add a flag --node-selector maybe a better way.
I will update it